backend/fms_core/resources/project_link_sample.py

```diff
@@ -0,0 +1,104 @@
+        except ValidationError as e:
+            errors.update(e.update_error_dict(errors).copy())
+        # Get sample object
+        try:
+            obj.sample = Sample.objects.get(name=data["Sample Name"],
+                                            container__barcode=data["Sample Container Barcode"],
+                                            coordinates=data["Sample Container Coord"])
+        except Sample.DoesNotExist:
+            obj.sample = None
+            errors["sample"] = ValidationError([f"No sample with name [{data['Sample Name']}] at those container coordinates."], code="invalid")
+        # Get project object
+        try:
+            obj.project = Project.objects.get(name=data["Project Name"])
+        except Project.DoesNotExist:
+            obj.project = None
+            errors["project"] = ValidationError([f"No project with name [{data['Project Name']}]"], code="invalid")
+
+        # Set object for deletion if REMOVE_ACTION
+        if SampleByProject.objects.filter(sample=obj.sample, project=obj.project).exists():
+            # Perform the desired action
+            if data["Action"] == REMOVE_ACTION:
+                    obj.deleted = True
+            elif data["Action"] == ADD_ACTION:
+                errors["link"] = ValidationError(
+                    [f"Sample [{data['Sample Name']}] is already associated to project [{data['Project Name']}]."],
+                    code="invalid")
+
+        else:
+            if data["Action"] == REMOVE_ACTION:
+                errors["link"] = ValidationError([f"Sample [{data['Sample Name']}] is not currently associated to project [{data['Project Name']}]."],
+                                                 code="invalid")
+
+        if errors:
+            raise ValidationError(errors)
+
+    def import_field(self, field, obj, data, is_m2m=False):
+        if field.column_name == "Action":
+            # Normalize Action attribute to be uppercase
+            data["Action"] = get_normalized_str(data, "Action").upper()
+        elif field.column_name == "Project Name":
+            data["Project Name"] = get_normalized_str(data, "Project Name")
+        elif field.column_name == "Sample Name":
+            data["Sample Name"] = get_normalized_str(data, "Sample Name")
+        elif field.column_name == "Sample Container Barcode":
+            data["Sample Container Barcode"] = get_normalized_str(data, "Sample Container Barcode")
+        elif field.column_name == "Sample Container Coord":
+            data["Sample Container Coord"] = get_normalized_str(data, "Sample Container Coord").upper()
+        super().import_field(field, obj, data, is_m2m)
+
+    def after_save_instance(self, instance, using_transactions, dry_run):
+        super().after_save_instance(instance, using_transactions, dry_run)
+        # If the instance is deleted (saved version), delete the instance
+        instance.refresh_from_db()
+        if instance.deleted:
```

Comment:
We could explain in a comment here why we actually skip model.delete() and that we manually set the deleted flag instead.  This has to do with preventing the creation of 2 saved versions before we delete the object (having import_export delete the object instead of saving it more complicated than this).